ops/testing.py

```diff
@@ -484,6 +486,24 @@ def get_pod_spec(self) -> (typing.Mapping, typing.Mapping):
         """
         return self._backend._pod_spec
 
+    def get_container_pebble_plan(
+            self, container_name: str
+    ) -> str:
+        """Return the current Plan that pebble is executing for the given container.
+
+        Args:
+            container_name: The simple name of the associated container
+        Return:
+            A list of service descriptions for the container.
+            Will return None if no pebble client exists for that container name.
+        """
+        socket_path = '/charm/containers/{}/pebble.socket'.format(container_name)
+        client = self._backend._pebble_clients.get(socket_path)
+        if client is None:
+            # TODO: jam 2021-04-19 Should this actually be a raise KeyError?
```

Comment:
Yeah, I'd lean towards raising the hard KeyError. I think that'd be a bug so it should be noisy.

Reply:
done. I wasn't sure at first, but since pebble client is initiated for all containers listed in metadata.yaml there isn't a need to initialize a plan first, so it makes sense to require it to be there.